### PR TITLE
WIP: Added grep support for cat APIs

### LINKED ISSUE #30245: Add grep support for cat APIs
We ``grep`` often the output of ``_cat`` API to find what we are looking for, why not add a parameter to do this directly?

``curl http://localhost:9200/_cat/indices?grep=events-``

This will make possible:
* Usage on a browser
* Share relevant URL with the grep info

### PR DESCRIPTION
**Pull request by sohaibiftikhar**
Resolves #30245

### CODE REVIEW

**Comment by sohaibiftikhar** [comment]
@nik9000 If you could take a cursory look when you have time and check if this is going in the right direction it would great. I can then add more tests and make the changes in the documentation.

File: server/src/main/java/org/elasticsearch/rest/action/cat/AbstractCatAction.java [hunk 1]
```diff
@@ -76,6 +84,7 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         }
     }
 
+    private String grepFilter;
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/rest/action/cat/AbstractCatAction.java]: These are singletons so you'd have to put this somewhere else or parse after rendering. You can use `responseParams` so the request will run.

File: server/src/main/java/org/elasticsearch/rest/action/cat/CatResponseTable.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import org.elasticsearch.rest.RestRequest;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CatResponseTable extends Table {
+
+    private boolean isValidRow;
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/rest/action/cat/CatResponseTable.java]: I'm fairly sure `Table` is only used for cat actions.

**Review by nik9000 — CHANGES_REQUESTED** [review]
I think it'd be better to do the grepping in `RestTable#buildTextPlainResponse` because you only render the strings one time and you get to grep on the whole line rather than the values one by one.

**Comment by jasontedor** [comment]
I do not agree with this feature being added to the codebase. Every feature we add is one that we have to maintain and have to consider maybe breaking in the future. The cat APIs are meant for the command line where grep is readily available?

**Comment by jasontedor** [comment]
Also this PR needs labels so that the appropriate area team is brought into the conversation.

**Comment by sohaibiftikhar** [comment]
@nik9000 Thanks for the review. Since the feature itself seems to be in question I will wait until this is decided.

**Comment by jasontedor** [comment]
Thanks for the PR @sohaibiftikhar yet we are not going to add this functionality to Elasticsearch.

**Pull request closed without merging**